### ISSUE TITLE
Potential bugs/bad style in example for _getdiskfree

### ISSUE THREAD

**Issue by ppelleti** [MicrosoftDocs/cpp-docs#3847]
Hi, the example for _getdiskfree() bothered me in a couple of ways:

## Use of non-literal format string in printf()

There are several places where the example calls printf() with a single argument.  For example, "printf(g_szBorder);" or "printf(szMsg);".  This is generally bad style, because the argument could contain a "%", which would then cause printf() to try to consume arguments which aren't there.

Technically, this should be safe for cases such as "printf(g_szBorder);" since the string literals are defined above and do not contain a "%".  Even though it's okay, it's a "code smell" for me personally, since the literal is located in a different place than the call to printf(), so it requires a little bit of effort to verify.

The most troubling case is "printf(szMsg);" since in that case the format string comes from FormatMessage().  I don't know how likely it is that FormatMessage() would return a string containing a "%" character, but I'm not aware of any guarantee that it won't happen.

Generally, I would recommend doing either:

```c
printf("%s", str);
```

or

```c
fputs(str, stdout);
```

in order to make it obvious that there won't be any misbehavior if str contains "%".

## Removal of terminating NUL byte from result of FormatMessage()

In the error case, after calling FormatMessage(), the code does the following:

```c
            szMsg[uLen+6] = '  ';
            szMsg[uLen+7] = '  ';
            szMsg[uLen+8] = '  ';
```

My understanding is that 6 and 7 are to remove the "\r\n" at the end of the string, and 8 is to remove the terminating NUL from the end of the string.

This will work just fine, as long as the length of the string returned by FormatMessage() plus 8 fits within the length of g_szLine[], because then the NUL byte at the end of g_szLine[] will terminate the string.

However, the call to FormatMessage() specifies a maximum length of 4100 characters:

```c
            uLen = FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM, NULL,
                            uErr, 0, szMsg+8, 4100, NULL);
```

Therefore, if the message is long, FormatMessage() could easily overwrite the NUL byte at the end of the version of g_szLine[] which has been copied into szMsg[].  If that happens, there does not appear to be any guarantee that a NUL byte will be reached before reaching the end of szMsg[] when printing the message.  (Since szMsg[] is uninitialized, beyond the first sizeof(g_szLine) bytes.)

A simple fix would be to add:

```c
         memset(szMsg, 0, sizeof(szMsg));
```

before the existing

```c
         memcpy(szMsg, g_szLine, sizeof(g_szLine));
```

although there would be other ways to fix it, as well.

Thanks!

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b3a07ec8-b631-20f0-c072-a03144c96487
* Version Independent ID: a49a780a-cb33-4ddd-8a7a-52bfea658e60
* Content: [_getdiskfree](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/getdiskfree?view=msvc-170)
* Content Source: [docs/c-runtime-library/reference/getdiskfree.md](https://github.com/Microsoft/cpp-docs/blob/main/docs/c-runtime-library/reference/getdiskfree.md)
* Product: **visual-cpp**
* Technology: **cpp-ucrt**
* GitHub Login: @TylerMSFT
* Microsoft Alias: **twhitney**

**Comment by ppelleti** [comment]
I've suggested one possible way to fix these two problems in my pull request #3851.

**Comment by TylerMSFT** [comment]
Hi @ppelleti,
Thank you for taking an interest in fixing the docs. I agree with your assessment but don't think the fix is to patch up the existing output code. I think it was a mistake to include all that table formatting code in the first place. I've removed it and replaced it with simple output which doesn't obscure the point of the topic, which was to demonstrate _getdiskfree(), not how to do print formatting.
The updates to the topic should be live by 5/12/2022.
